### PR TITLE
added middleware for autherizing roles

### PR DESCRIPTION
**Pull request by ludbek**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -144,8 +144,33 @@ function verifyUserSignature(options){
   }
 }
 
+// express authorization middleware
+// Usage: route('xxxxxxxxxx', canBeAccessedBy('a-user'), handler)
+// Usage: route('xxxxxxxxxx', canBeAccessedBy(['a-user', 'b-user']), handler)
```
Comment (joshuapaling) [inline index.js]: would be clearer if comment referred to actual examples - eg `canBeAccessedBy(['admin-user', 'employee'])`

File: test/index_test.js [hunk 2]
```diff
@@ -193,4 +225,46 @@ describe('auth', function() {
...
+      chai.request(app)
+      .get("/middleware/allow-multiple-roles")
+      .set('Cookie', authed_token)
+      .end(function(error, response) {
+        expect(response.statusCode).to.equal(200);
+        done();
+      });
+    })
+
+    it("should allow single role", function (done) {
```
Comment (forkfork) [inline test/index_test.js]: text

File: test/index_test.js [hunk 3]
```diff
@@ -193,4 +225,46 @@ describe('auth', function() {
...
+      chai.request(app)
+      .get("/middleware/disallow-single-role")
+      .set('Cookie', authed_token)
+      .end(function(error, response) {
+        expect(response.statusCode).to.equal(403);
+        done();
+      });
+    })
+
+    it("should allow multiple roles", function(done) {
```
Comment (forkfork) [inline test/index_test.js]: text

**Comment by ludbek** [comment]
@forkfork sorry i got confused, the text is correct. The endpoints should return text `allowed` when accessed by proper user.

**Pull request merged**